server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -301,6 +302,25 @@ export function configureWebSocketServices(
 				throw new NetworkError(403, "Must provide an authorization token");
 			}
 
+			// Check if current cluster is in draining
+			if (clusterDrainingChecker) {
+				let clusterInDraining = false;
+				try {
+					clusterInDraining = await clusterDrainingChecker.isClusterDraining();
+				} catch (error) {
+					Lumberjack.error(
+						"Failed to get cluster draining status. Will allow requests to proceed.",
+						undefined,
+						error,
+					);
+					clusterInDraining = false;
+				}
+				if (clusterInDraining) {
+					// TODO: add a new error class
+					throw new NetworkError(503, "Cluster is not available. Please retry later.");
+				}
```

Comment:
Let's generate a log in addition to throwing the error?

Reply:
Fixed